myst_nb/parser.py

```diff
@@ -21,6 +30,92 @@ class NotebookParser(MystParser):
     config_section_dependencies = ("parsers",)
 
     def parse(self, inputstring, document):
+
+        # de-serialize the notebook
+        ntbk = nbf.reads(inputstring, nbf.NO_CONVERT)
+
+        # This is a contaner for top level markdown tokens
+        # which we will add to as we walk the document
+        mkdown_tokens = []  # type: list[BlockToken]
+
+        # First we ensure that we are using a 'clean' global context
+        # for parsing, which is setup with the MyST parsing tokens
+        # the logger will report on duplicate link/footnote definitions, etc
+        parse_context = ParseContext(
+            find_blocks=SphinxNBRenderer.default_block_tokens,
+            find_spans=SphinxNBRenderer.default_span_tokens,
+            logger=SPHINX_LOGGER,
+        )
+        set_parse_context(parse_context)
+
+        for cell_index, nb_cell in enumerate(ntbk.cells):
+
+            # Skip empty cells
+            if len(nb_cell["source"].strip()) == 0:
+                continue
+
+            # skip cells tagged for removal
+            tags = nb_cell.metadata.get("tags", [])
+            if "remove_cell" in tags:
+                continue
+
+            if nb_cell["cell_type"] == "markdown":
+
+                # we add the document path and cell index
+                # to the source lines, so they can be included in the error logging
+                # NOTE: currently the logic to report metadata is not written
+                # into SphinxRenderer, but this will be introduced in a later update
+                lines = SourceLines(
+                    nb_cell["source"],
+                    uri=document["source"],
+                    metadata={"cell_index": cell_index},
+                    standardize_ends=True,
+                )
+
+                # parse the source markdown text;
+                # at this point span/inline level tokens are not yet processed, but
+                # link/footnote definitions are collected/stored in the global context
+                mkdown_tokens.extend(tokenize_block(lines))
+
+                # TODO think of a way to implement the previous
```

Comment:
Is this a regression then? (currently `hide_input` etc does work, so will this remove that functionality?)

Reply:
it seems below that we *do* handle tags in the Renderer, so maybe this comment just needs to be removed?

Reply:
This comment is **only** referring to `hide_input` tags for markdown cells. But as we discussed, we can now handle this by directives. So I can remove this comment if you are happy with losing that functionality.

Reply:
i.e. we no longer implement this code: https://github.com/ExecutableBookProject/MyST-NB/blob/2207e30a63928c4d96c46a395512608595735fa5/myst_nb/parser.py#L89